src/software/simulated_tests/validation_functions/robots_in_friendly_half_validation.cpp

```diff
@@ -0,0 +1,15 @@
+#include "software/simulated_tests/validation_functions/robots_in_friendly_half_validation.h"
+
+#include <gtest/gtest.h>
+
+void robotsInFriendlyHalf(std::shared_ptr<World> world_ptr,
+                          ValidationCoroutine::push_type& yield)
+{
+    for (auto robot : world_ptr->friendlyTeam().getAllRobots())
+    {
+        if (world_ptr->field().pointInEnemyHalf(robot.position()))
```

Comment:
```suggestion
        if (!world_ptr->field().pointInFriendlyHalf(robot.position()))
```
This would catch robots going out of bounds and is more precise to the name

Reply:
Updated!